pkg/controller/serviceaccount/tokens_controller.go

```diff
@@ -156,23 +156,23 @@ type TokensController struct {
 }
 
 // Run runs controller blocks until stopCh is closed
-func (e *TokensController) Run(workers int, stopCh <-chan struct{}) {
+func (e *TokensController) Run(ctx context.Context, workers int) {
 	// Shut down queues
 	defer utilruntime.HandleCrash()
 	defer e.syncServiceAccountQueue.ShutDown()
 	defer e.syncSecretQueue.ShutDown()
 
-	if !cache.WaitForNamedCacheSync("tokens", stopCh, e.serviceAccountSynced, e.secretSynced) {
+	if !cache.WaitForNamedCacheSync("tokens", ctx.Done(), e.serviceAccountSynced, e.secretSynced) {
 		return
 	}
 
-	klog.V(5).Infof("Starting workers")
+	klog.FromContext(ctx).V(5).Info("Starting workers")
```

Comment:
Nit: just for readability it's easier to grasp if you separate this line into two:
```golang
logger := klog.FromContext(ctx)
logger.V(5)...
```
especially that you're using this here and a few lines below.

Reply:
nice catch! addressed in this commit https://github.com/kubernetes/kubernetes/pull/114471/commits/8b67fbc39de5f6729e7c7edf2e81ae6bfa9e5c47